src/UI/templates/js/Button/button.js

```diff
@@ -28,8 +30,55 @@ il.UI.button = il.UI.button || {};
 			});
 		};
 
+		/* toggle button */
+		var handleToggleClick = function (event, id, on_url, off_url, signals) {
+			var b = $("#" + id);
+			var pressed = b.attr("aria-pressed");
+			for (var i = 0; i < signals.length; i++) {
+				var s = signals[i];
+				if (s.event === "click" ||
+					(pressed === "true" && s.event === "toggle_on") ||
+					(pressed !== "true" && s.event === "toggle_off")
+				) {
+					$(b).trigger(s.signal_id, {
+						'id' : s.signal_id,
+						'event' : s.event,
+						'triggerer' : b,
+						'options' : s.options});
+				}
```

Comment:
could we not be more explicit and only fire on and off, if really/off is fired? Meaning leaving the decision about on and off to the Toggle Button and not to the components receiving the signal? Possible that I am missing something here.

Reply:
The toggle button makes the decision whether toggle_on or toggle_off signals are fired. The consumer makes the decision which signals are attached to the toggle_on or toggle_off events.

Reply:
> The toggle button makes the decision whether toggle_on or toggle_off signals are fired. The consumer makes the decision which signals are attached to the toggle_on or toggle_off events.

Sounds good, however, at the current state of the contribution, all signals are fired on click, right? As said, probably if missed something. If only the right once are fired, I am fine.